From a1494209e2eb9d64cbb45dd2b64f126184378496 Mon Sep 17 00:00:00 2001
From: Simon Cook <simon.cook@embecosm.com>
Date: Thu, 25 Sep 2014 18:06:29 +0100
Subject: [PATCH 16/33] Fix software cache flag issues

	* config/epiphany/epiphany.h (EPIPHANY_LIBRARY_EXTRA_SPEC): Don't
	build libraries with -fpic by default.
	* config/epiphany/epiphany.opt: Fix CC1 warning when
	-fsoftware-cache enabled.
---
 gcc/ChangeLog.epiphany           | 7 +++++++
 gcc/config/epiphany/epiphany.h   | 2 +-
 gcc/config/epiphany/epiphany.opt | 1 +
 3 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/gcc/ChangeLog.epiphany b/gcc/ChangeLog.epiphany
index f66a52b0ff4..572b6af3a67 100644
--- a/gcc/ChangeLog.epiphany
+++ b/gcc/ChangeLog.epiphany
@@ -1,3 +1,10 @@
+2014-09-25  Simon Cook  <simon.cook@embecosm.com>
+
+	* config/epiphany/epiphany.h (EPIPHANY_LIBRARY_EXTRA_SPEC): Don't
+	build libraries with -fpic by default.
+	* config/epiphany/epiphany.opt: Fix CC1 warning when
+	-fsoftware-cache enabled.
+
 2014-09-23  Simon Cook  <simon.cook@embecosm.com>
 
 	* config/epiphany/epiphany.h (ENDFILE_SPEC): Add cachemanager.o
diff --git a/gcc/config/epiphany/epiphany.h b/gcc/config/epiphany/epiphany.h
index 3a63160b53e..8f6b27784a1 100644
--- a/gcc/config/epiphany/epiphany.h
+++ b/gcc/config/epiphany/epiphany.h
@@ -60,7 +60,7 @@ along with GCC; see the file COPYING3.  If not see
 #define CC1_SPEC "%{fsoftware-cache:-fpic}"
 
 #define EPIPHANY_LIBRARY_EXTRA_SPEC \
-  "-ffixed-r41 -ffixed-r41 -ffixed-r42 -ffixed-r43 -fpic"
+  "-ffixed-r40 -ffixed-r41 -ffixed-r42 -ffixed-r43"
 
 /* In the "spec:" rule,, t-epiphany changes this to epiphany_library_stub_spec
    and epiphany_library_extra_spec, respectively.  */
diff --git a/gcc/config/epiphany/epiphany.opt b/gcc/config/epiphany/epiphany.opt
index 9055f0e62c7..549a4a095f5 100644
--- a/gcc/config/epiphany/epiphany.opt
+++ b/gcc/config/epiphany/epiphany.opt
@@ -148,4 +148,5 @@ EnumValue
 Enum(m1reg) String(r63) Value(63)
 
 fsoftware-cache
+Target Mask(SOFT_CACHE)
 Use software cache
-- 
2.31.1

